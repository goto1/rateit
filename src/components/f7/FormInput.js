import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Icon from "./Icon";

const InputWrapper = styled.div`margin-left: 15px;`;

const FormInput = ({ icon, type, name, placeholder, onChange }) =>
  <div className="item-content">
    <div className="item-media">
      <Icon material={icon} />
    </div>
    <InputWrapper className="item-input">
      <input
        type={type}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
      />
    </InputWrapper>
  </div>;

FormInput.propTypes = {
  icon: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default FormInput;
