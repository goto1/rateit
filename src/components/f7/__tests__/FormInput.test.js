import React from "react";
import FormInput from "../FormInput";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

describe("FormInput", () => {
  const mockFn = jest.fn();

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <FormInput
          icon="account_box"
          type="text"
          name="username"
          placeholder="Type your username"
          onChange={mockFn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls a function when input changes", () => {
    const wrapper = shallow(
      <FormInput
        icon="account_box"
        type="text"
        name="username"
        placeholder="Type your username"
        onChange={mockFn}
      />
    );
    wrapper.find("input").simulate("change");

    expect(mockFn.mock.calls.length).toBe(1);
  });
});
