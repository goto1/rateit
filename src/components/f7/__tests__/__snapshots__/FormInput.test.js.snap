// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component renders correctly 1`] = `
.bSLHOK {
  margin-left: 15px;
}

<div
  className="item-content"
>
  <div
    className="item-media"
  >
    <i
      className="material-icons icon"
    >
      account_box
    </i>
  </div>
  <div
    className="item-input sc-bdVaJa bSLHOK"
  >
    <input
      name="username"
      onBlur={undefined}
      onChange={[Function]}
      placeholder="Type your name"
      type="text"
    />
  </div>
</div>
`;
